web/packages/teleport/src/Clusters/ManageCluster/ManageCluster.tsx

```diff
@@ -0,0 +1,211 @@
+/**
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { Link } from 'react-router-dom';
+
+import useAttempt, { Attempt } from 'shared/hooks/useAttemptNext';
+
+import { MultiRowBox, Row } from 'design/MultiRowBox';
+import Flex from 'design/Flex';
+import * as Icons from 'design/Icon';
+import Text, { H2 } from 'design/Text';
+import { Indicator } from 'design/Indicator';
+
+import Box, { BoxProps } from 'design/Box';
+
+import { useParams } from 'react-router';
+
+import { LoadingSkeleton } from 'shared/components/UnifiedResources/shared/LoadingSkeleton';
+
+import { ShimmerBox } from 'design/ShimmerBox';
+import { Alert } from 'design/Alert';
+
+import {
+  FeatureBox,
+  FeatureHeader,
+  FeatureHeaderTitle,
+} from 'teleport/components/Layout';
+import { useTeleport } from 'teleport/index';
+import cfg from 'teleport/config';
+import { useNoMinWidth } from 'teleport/Main';
+import { ClusterInfo } from 'teleport/services/clusters';
+
+export function ManageCluster() {
+  const [cluster, setCluster] = useState<ClusterInfo>(null);
+  const { attempt, run } = useAttempt();
```

Comment:
I think the newer fetch functions in the front end are now using `useAsync` (`shared/hooks/useAsync`) instead of `useAttempt`.

Reply:
Yeah that is the way to do it. To use it in `useEffect`, you will need `useCallback` to make the signature of the `useasync` function stable.